### ISSUE TITLE
Attribute #in returning an actual Arel::Attributes::Attribute

### ISSUE THREAD

**Issue by rapides** [rzane/baby_squeel#61]
#### Issue

There is error when subquery is empty.

#### Reproduction

```ruby
[4] pry(main)> User.where.has { id.in(User.none.select(:id)) | id.eq(5) }
NoMethodError: undefined method `|' for #<Arel::Nodes::In:0x007f99635f5fb0>
from (pry):6:in `block in <main>'

[2] pry(main)> User.where.has { id.in(User.none.select(:id)) }
  User Load (2.2ms)  SELECT "users".* FROM "users" WHERE "users"."id" IN ()
=> #<User::ActiveRecord_Relation:0x3fccb1f90084>
```

**Comment by rzane** [comment]
hi @rapides, thanks for reporting this! I just released 1.1.3.

**Comment by rapides** [comment]
hi @rzane 

I think that your fix doesn't cover this issue fully yet. Your fix should prevent against a dangerous situation where nested query returns empty table. Currently implemented solution may provide many unexpected random errors.

```
[4] pry(main)> User.where.has { id.in(User.none.select(:id)) }.explain
  User Load (1.0ms)  SELECT "users".* FROM "users" WHERE "users"."id" IN ()
ActiveRecord::StatementInvalid: PG::SyntaxError: ERROR:  syntax error at or near ")"
LINE 1: SELECT "users".* FROM "users" WHERE "users"."id" IN ()
                                                             ^
: SELECT "users".* FROM "users" WHERE "users"."id" IN ()
from /Users/rapide/.rvm/gems/ruby-2.3.0@bttf/gems/activerecord-4.2.8/lib/active_record/connection_adapters/postgresql_adapter.rb:592:in `async_exec'
```
I won't want to check every time if my subquery return something.

Looking forward for some solution.

**Comment by rzane** [comment]
Yeah, there should definitely be a graceful solution for this. Nice catch!

Here's how Active Record handles the issue:

```ruby
Post.where(author_id: Author.none.select(:id))
# Post Load (0.2ms)  SELECT "posts".* FROM "posts" WHERE "posts"."author_id" IN (SELECT "authors"."id" FROM "authors" WHERE (1=0))
```

**Comment by rzane** [comment]
Fixed in #63

**Comment by rapides** [comment]
Many thanks! Is it possible to release that in separate version?

**Comment by rzane** [comment]
I just released 1.1.4.
